QA-4765-implementazione-test-incaricato

leonardobaionidgs opened the pull request.

leonardobaionidgs committed:
Defined first structure for SRS Incaricato

Vincenzo-Massaro committed:
Modified bffApi.yml link in pom.xml

Modified pointing to the APIs set bffApi.yml and updated Java objects accordingly

Vincenzo-Massaro committed:
Added delegation consumer APIs and Cucumber step, error ongoing

Running of TC_INCARICATO_45 produces the following error, analysis in progress:
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot POST /backend-for-frontend/1.0/tenants/delegatedConsumer</pre>
</body>
</html>

Vincenzo-Massaro committed:
Added consumer delegation creation support

Added consumer delegation creation services, configuration and steps.
Concerning the error regarding TC_INCLOSED_45 mentioned in the previous commit, it is due to the fact that the endpoints used are not yet up-and-running.

mmancini95 committed:
QA-4765: added step for consumer rejection

Vincenzo-Massaro committed:
First implementation of some Incaricato tests

- refactored tests from TC_INCARICATO_45 to TC_INCARICATO_47
- implemented from TC_INCARICATO_48 to TC_INCARICATO_51
- added steps distinguishing between delegate and delegator
- expanded DelegationCommonContext.java
- moved DelegationParameterTypes.java and DelegationRole.java from DelegationCreateStep.java to their own files

Vincenzo-Massaro committed:
Implemented and formally fixed to TC_INCARICATO_56

Vincenzo-Massaro committed:
Implemented agreement (fruizione) section of tests

- implemented Incaricato tests from TC_INCARICATO_57 to TC_INCARICATO_60
- removed TC_INCARICATO_61 because is no longer specified in the SRS

Vincenzo-Massaro committed:
First implementation of other Incaricato tests

Implemented TC_INCARICATO_62, TC_INCARICATO_67, TC_INCARICATO_72, TC_INCARICATO_73, TC_INCARICATO_74

Vincenzo-Massaro committed:
Extended steps in AgreementCommonSteps.java

Vincenzo-Massaro committed:
First impl. of TC_INCARICATO_63

Vincenzo-Massaro committed:
First impl. of TC_INCARICATO_64 and TC_INCARICATO_64_BIS

Vincenzo-Massaro committed:
First impl. of TC_INCARICATO_65 and TC_INCARICATO_66

Vincenzo-Massaro committed:
First impl. of TC_INCARICATO_68, TC_INCARICATO_69 and TC_INCARICATO_71

mmancini95 committed:
QA-4765: added the correct steps to create an e-service

mmancini95 committed:
Merge remote-tracking branch 'origin/QA-4765-implementazione-test-incaricato' into QA-4765-implementazione-test-incaricato

Vincenzo-Massaro committed:
First impl. of TC_INCARICATO_70, proposals for improvement in incaricato-refac.feature

leonardobaionidgs committed:
Extended implementation of Incaricato tests

Vincenzo-Massaro committed:
Finished impl. of @TC_INCARICATO_58_BIS

Vincenzo-Massaro commented on interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/agreement/AgreementCreationStep.java:
```diff
@@ -61,6 +61,13 @@ public void delegationNotExistRequestForServiceAlreadySubmittedAndPendingApprova
         agreementProcessRequest(tenantType, delegationId);
     }
 
+    @Given("il {delegationRole} ha già creato e inviato una richiesta di fruizione indicando la delega dell'ente terzo")
```
Diversamente dal flusso normale qui si dà modo di specificare una delega che esiste ma che NON appartiene al delegato

Vincenzo-Massaro commented on interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/common/DelegationCommonContext.java:
```diff
@@ -18,6 +18,10 @@ public class DelegationCommonContext {
     private String delegatorTenant;
     private String delegateTenant;
 
+    /* 06/02/2025 useful in those test cases where it is necessary to have a third-party
```
Il campo dà modo di specificare una delega che non intercorre tra il delegante e il delegato, ma tra il delegante e un ente terzo

Vincenzo-Massaro commented on interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/delegate/DelegationCreateStep.java:
```diff
@@ -101,9 +101,39 @@ public void givenDelegatingTenantHasRequestedDelegation() {
     @Given("l'ente delegante ha inoltrato una richiesta di delega in fruizione all'ente delegato")
     public void givenConsumerDelegatingTenantHasRequestedDelegation() {
-        String delegatingTenantToken = identityService.getToken(sharedStepsContext.getDelegationCommonContext().getTenantBy(DELEGATING), null);
+        String delegatorTenant = sharedStepsContext.getDelegationCommonContext().getTenantBy(DELEGATING);
```
Aggiunti steps in cui si permette si specificare un ente terzo per la delega, e riarrangiati leggermente i metodi esistenti per  permettere maggiore versatilità e riutilizzabilità del codice

Vincenzo-Massaro commented on interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/delegate/DelegationCreateStep.java:
```diff
@@ -165,15 +195,24 @@ public void delegationIsPresent() {
     }
 
-    private void createDelegate(String tenantType, BiFunction<String, DelegationSeed, CreatedResource> delegationCreator) {
+    private void createDelegate(
```
Implementazione rimaneggiata così che richiami semplicemente il nuovo metodo sottostante

Vincenzo-Massaro commented on interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/delegate/DelegationCreateStep.java:
```diff
@@ -165,15 +195,24 @@ public void delegationIsPresent() {
+        this.createDelegate(tenantType, delegationCreator, DelegationProxy.ofMainDelegation(sharedStepsContext.getDelegationCommonContext()));
+    }
+
+    private void createDelegate(
```
Aggiunto parametro DelegationProxy per permettere il setting e il getting del giusto delegation ID in maniera polimorfica

Vincenzo-Massaro commented on interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/delegate/DelegationProxy.java:
L'introduzione della variabile finora chiamata "ente terzo" ha portato un secondo delegation ID oltre l'unico precedentemente previsto. In alcuni punti del codice si ha bisogno di poter interagire con l'uno o l'altro a seconda del contesto: queste classi servono proprio a indicare, in maniera trasparente, quale delegation ID usare. Si ha un esempio nei metodi:
- it.pagopa.pn.interop.cucumber.steps.delegate.DelegationCreateStep#givenConsumerDelegatingTenantHasRequestedDelegation(java.lang.String, java.lang.String)
- it.pagopa.pn.interop.cucumber.steps.delegate.DelegationCreateStep#givenConsumerDelegatingTenantHasRequestedAuxDelegation(java.lang.String, java.lang.String)

Vincenzo-Massaro committed:
Impl. of @TC_INCARICATO_58_TRIS

Vincenzo-Massaro committed:
Finish impl. of TC_INCARICATO_87 and TC_INCARICATO_88, collapsed 89-90-91 -> 89

Vincenzo-Massaro committed:
Alignment with new bff APIs version in develop

Vincenzo-Massaro committed:
Corrected test id locations in incaricato.feature and introduced IncaricatoTest.java

Vincenzo-Massaro committed:
Ongoing 'Background' clause introduction

Vincenzo-Massaro committed:
Ongoing TC_INCARICATO_47 fix

mmancini95 committed:
QA-4765: populated eservice delegable flags before publishing the eservice

Vincenzo-Massaro committed:
Merge remote-tracking branch 'origin/QA-4765-implementazione-test-incaricato' into QA-4765-implementazione-test-incaricato

mmancini95 committed:
QA-4765: used token previously stored in sharedStepsContext in the authAndConsumerDelegation method

Vincenzo-Massaro committed:
Fixed some e-services creation step

Vincenzo-Massaro committed:
Merge remote-tracking branch 'origin/QA-4765-implementazione-test-incaricato' into QA-4765-implementazione-test-incaricato

mmancini95 committed:
QA-4765: updated Incaricato tests with minor fixes and introduced the parameterType for the agreement approval policy

Vincenzo-Massaro committed:
QA-4765: revisited agreement creation and reworked TC_INCARICATO_57
- expanded and revisited steps for agreement creation
- TC_INCARICATO_57: reformulated so that it only tests positive cases, while negative cases have been moved to an independent test

mmancini95 committed:
QA-4765: fixed test cases up to TC_INCARICATO_70

Vincenzo-Massaro committed:
QA-4765: ongoing fix of TC_INCARICATO_58_xxx

Vincenzo-Massaro committed:
QA-4765: fixed TC_INCARICATO_58_xxx

- TC_INCARICATO_58_BIS: the previous version tested an impossible condition, in that a delegator cannot appoint more than one delegate for the same e-service. Modified so that it tests the impossibility of appointing more than one delegate for the same e-service
- TC_INCARICATO_58_TRIS temporarily deactivated as tenant 'Private' cannot perform delegation requests (currently no other tenant available)

Vincenzo-Massaro committed:
Merge remote-tracking branch 'origin/QA-4765-implementazione-test-incaricato' into QA-4765-implementazione-test-incaricato-TC_INCARICATO_57

# Conflicts:
#	interop-qa-tests/src/test/resources/it/pagopa/pn/cucumber/incaricato/consumer.feature

Vincenzo-Massaro committed:
QA-4765: ongoing fix TC_INCARICATO_85

Vincenzo-Massaro committed:
QA-4765: ongoing fix TC_INCARICATO_85

Vincenzo-Massaro committed:
QA-4765: ongoing fix TC_INCARICATO_85

mmancini95 committed:
QA-4765: updated tests and aligned with SRS

mmancini95 committed:
Merge remote-tracking branch 'origin/QA-4765-implementazione-test-incaricato' into QA-4765-implementazione-test-incaricato

Vincenzo-Massaro committed:
QA-4765: ongoing fix TC_INCARICATO_87

Vincenzo-Massaro committed:
QA-4765: fixed TC_INCARICATO_87

Vincenzo-Massaro committed:
QA-4765: ongoing fix of TC_INCARICATO_89

Vincenzo-Massaro committed:
QA-4765: fixed TC_INCARICATO_89, TC_INCARICATO_90

Vincenzo-Massaro committed:
QA-4765: fixed TC_INCARICATO_77

Vincenzo-Massaro committed:
QA-4765: added TC_INCARICATO_47_BIS

mmancini95 committed:
QA-4765: removed wrong tests and fixed TC_INCARICATO_78

mmancini95 committed:
Merge remote-tracking branch 'origin/QA-4765-implementazione-test-incaricato' into QA-4765-implementazione-test-incaricato

Vincenzo-Massaro committed:
QA-4765: fixed TC_INCARICATO_58_TRIS

Vincenzo-Massaro committed:
Merge remote-tracking branch 'origin/QA-4765-implementazione-test-incaricato' into QA-4765-implementazione-test-incaricato-TC_INCARICATO_57

# Conflicts:
#	interop-qa-tests/src/test/resources/it/pagopa/pn/cucumber/incaricato/consumer.feature

mmancini95 committed:
Merge pull request #660 from pagopa/QA-4765-implementazione-test-incaricato-TC_INCARICATO_57

QA-4765: revisited agreement creation and reworked TC_INCARICATO_57

Vincenzo-Massaro committed:
QA-4765: marked tests to fix

Vincenzo-Massaro committed:
QA-4765: minor fix to TC_INCARICATO_54

mmancini95 committed:
QA-4765: fix test capofila

Vincenzo-Massaro committed:
QA-4765: quick scenario fixes

mmancini95 committed:
QA-4765: removed exclude tag

mmancini95 committed:
Merge remote-tracking branch 'origin/QA-4765-implementazione-test-incaricato' into QA-4765-implementazione-test-incaricato

Vincenzo-Massaro committed:
QA-4765: aligned TC_INCARICATO_51

mmancini95 committed:
QA-4765: fixed test case TC_INCARICATO_67 e TC_INCARICATO_78

mmancini95 committed:
QA-4765: removed @wait_for_fix tag

Vincenzo-Massaro committed:
Merge remote-tracking branch 'origin/QA-4765-implementazione-test-incaricato' into QA-4765-implementazione-test-incaricato

mmancini95 committed:
Merge branch 'QA-4765-implementazione-test-incaricato' into QA-4765-fix-test-capofila

mmancini95 committed:
QA-4765: removed some comments in consumer.feature

mmancini95 committed:
QA-4765: fixed test TC_INCARICATO_67 and TC_INCARICATO_71

mmancini95 committed:
QA-4765: updated status code for test TC_INCARICATO_52

mmancini95 committed:
QA-4765: removed comment

mmancini95 committed:
QA-4765: updated status code returned from test TC_INCARICATO_73

mmancini95 committed:
Merge branch 'QA-4765-implementazione-test-incaricato' into QA-4765-fix-test-capofila

mmancini95 committed:
QA-4765: added tests for the endpoint getConsumerDelegatorsWithAgreements, getConsumerDelegators, getConsumerDelegatedEservices

Vincenzo-Massaro committed:
QA-4765: improved TC_INCARICATO_57 last assertions

mmancini95 committed:
QA-4765: fixed TC_INCARICATO_57

mmancini95 committed:
QA-4765: removed a no longer required method

mmancini95 committed:
Merge pull request #661 from pagopa/QA-4765-fix-test-capofila

Qa 4765 fix test capofila

mmancini95 committed:
QA-4765: Merge branch 'develop' into feature/QA-4765-implementazione-test-incaricato

# Conflicts:
#	interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/delegate/DelegationCommonStep.java
#	interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/delegate/DelegationListingStep.java

mmancini95 committed:
QA-4765: updated the tenants API based on the new changes introduced to the endpoints

mmancini95 committed:
QA-4765: Merge branch 'develop' into QA-4765-implementazione-test-incaricato

# Conflicts:
#	interop-qa-tests/pom.xml
#	interop-qa-tests/src/main/java/it/pagopa/interop/agreement/service/impl/AgreementClientImpl.java
#	interop-qa-tests/src/main/java/it/pagopa/interop/delegate/service/impl/ProducerDelegationsApiClientImpl.java
#	interop-qa-tests/src/main/java/it/pagopa/interop/tenant/service/impl/TenantsApiClientImpl.java
#	interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/DataPreparationService.java
#	interop-qa-tests/src/test/java/it/pagopa/pn/interop/cucumber/steps/InteropCucumberSpringIntegration.java

mmancini95 committed:
QA-4765: removed consumer-refac.feature

mmancini95 committed:
QA-4765: removed methods and classes not used
